R/data_to_wide.R

```diff
@@ -412,6 +401,46 @@ data_to_wide <- function(data,
 }
 
 
+#' Fills missing values in the widened newley crated columns in `values_from`
+#' with `values_fill`
+#'
+#' @noRd
+
+.fill_missings <- function(x, values_from, values_fill, verbose = TRUE) {
+  if (!is.null(values_fill)) {
+    if (length(values_fill) == 1L) {
+      if (all(vapply(values_from, function(i) is.numeric(x[[i]]), logical(1)))) {
+        if (is.numeric(values_fill)) {
+          x <- convert_na_to(x, replace_num = values_fill)
+        } else {
+          insight::format_error(paste0("`values_fill` must be of type numeric."))
+        }
+      } else if (all(vapply(values_from, function(i) is.character(x[[i]]), logical(1)))) {
+        if (is.character(values_fill)) {
+          x <- convert_na_to(x, replace_char = values_fill)
+        } else {
+          insight::format_error(paste0("`values_fill` must be of type character."))
+        }
+      } else if (all(vapply(values_from, function(i) is.factor(x[[i]]), logical(1)))) {
+        if (is.factor(values_fill)) {
+          x <- convert_na_to(x, replace_fac = values_fill)
+        } else {
+          insight::format_error(paste0("`values_fill` must be of type factor."))
+        }
+      } else if (verbose) {
+        insight::format_error(
+          "No missing values were filled, because either `values_from` contains variables of different types, or the type of `values_fill` is not supported."
+        )
+      }
+    } else if (verbose) {
+      insight::format_error("`values_fill` must be of length 1.")
+    }
```

Comment:
I'll properly review later today but can this be unnested by one level by doing 

```r
if (length(values_fill) != 1L && verbose) {
      insight::format_error("`values_fill` must be of length 1.")
}
```
? I originally wrote it but I guess this simplification is possible.